src/PIL/Image.py

```diff
@@ -1565,6 +1565,8 @@ def getpixel(self, xy):
         self.load()
         if self.pyaccess:
             return self.pyaccess.getpixel(xy)
+        if isinstance(xy, list):
+            xy = tuple(xy)
         return self.im.getpixel(xy)
```

Comment:
Unconditional casting is faster and makes possible [other types which was asked](https://github.com/python-pillow/Pillow/issues/7352#issuecomment-1693088128).

```python
In [1]: a = (1, 2)

In [3]: %timeit _ = tuple(a)
29.4 ns ± 0.059 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [4]: %%timeit
   ...: if isinstance(a, list):
   ...:     _ = tuple(a)
51.9 ns ± 0.0765 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [2]: b = [1, 2]

In [5]: %timeit _ = tuple(b)
42.8 ns ± 0.0378 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [6]: %%timeit
   ...: if isinstance(b, list):
   ...:     _ = tuple(b)
83.4 ns ± 0.235 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```

```suggestion
        return self.im.getpixel(tuple(xy))
```

Reply:
In main at the moment, `im.getpixel(0)` raises `TypeError: argument must be sequence of length 2`

Your suggestion would raise instead `TypeError: 'int' object is not iterable`

I was trying to avoid that change. Perhaps it is not a significant concern however.